storage/remote/client.go

```diff
@@ -162,6 +165,9 @@ func (c *Client) Store(ctx context.Context, req []byte) error {
 	httpReq.Header.Set("Content-Type", "application/x-protobuf")
 	httpReq.Header.Set("User-Agent", UserAgent)
 	httpReq.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
+	for k, v := range c.headers {
+		httpReq.Header.Set(k, v)
```

Comment:
Can you move that before the other headers? It would be unsafe to overwrite the other headers.

Reply:
Thank you, I have just updated my branch according to your comments.